### PR TITLE
[Patch 1/2] :: Add support to Marketplace endpoints

### PR DESCRIPTION
**Pull request by PauloMigAlmeida**
# Description 
Add support to Marketplace endpoints

Endpoints added: 
- https://developer.github.com/v3/apps/marketplace/#list-all-plans-for-your-marketplace-listing
- https://developer.github.com/v3/apps/marketplace/#list-all-github-accounts-user-or-organization-on-a-specific-plan

In the best interest of keeping this PR readable (too many lines due to the integration tests files), I will submit a second PR to add the remaining endpoints related to GitHub Marketplace.

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [X] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [X] Add JavaDocs and other comments
- [X] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [X] Run `mvn install site` locally. This may reformat your code, commit those changes. If this command doesn't succeed, your change will not pass CI.

### CODE REVIEW

**Comment by PauloMigAlmeida** [comment]
@bitwiseman this PR is ready for your review :)

**Comment by bitwiseman** [comment]
@PauloMigAlmeida 
I'm on vacation, I'll look at this shortly.

File: src/main/java/org/kohsuke/github/GHMarketplacePendingChange.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+/**
+ * A Github Marketplace purchase pending change.
+ *
+ * @author Paulo Miguel Almeida
+ * @see GHMarketplaceListAccountBuilder#retrieve()
+ */
+public class GHMarketplacePendingChange {
+    private GitHub root;
+    private long id;
+    @JsonProperty("unit_count")
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHMarketplacePendingChange.java]:
```suggestion
```

File: src/main/java/org/kohsuke/github/GHMarketplacePendingChange.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    private Long unitCount;
+    @SuppressFBWarnings(value = "UWF_UNWRITTEN_FIELD", justification = "Field comes from JSON deserialization")
+    private GHMarketplacePlan plan;
+    @JsonProperty("effective_date")
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHMarketplacePendingChange.java]:
```suggestion
```

File: src/main/java/org/kohsuke/github/GHMarketplacePlan.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+ * A Github Marketplace plan.
+ *
+ * @author Paulo Miguel Almeida
+ * @see GitHub#listMarketplacePlans()
+ */
+public class GHMarketplacePlan {
+
+    private GitHub root;
+    private String url;
+    @JsonProperty("accounts_url")
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHMarketplacePlan.java]:
```suggestion
```

And the same for most of the other JsonProperties here.

File: src/main/java/org/kohsuke/github/GHMarketplaceAccount.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.net.URL;
+
+/**
+ * A Github Marketplace Account.
+ *
+ * @author Paulo Miguel Almeida
+ * @see GHMarketplaceListAccountBuilder#retrieve()
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHMarketplaceAccount.java]:
```suggestion
 * @see GHMarketplaceListAccountBuilder#createRequest()
```

Changed this in the `GitHub` class, should try to stay consistent.

File: src/main/java/org/kohsuke/github/GHMarketplaceListAccountBuilder.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ * @see GHMarketplacePlan#listAccounts()
+ */
+public class GHMarketplaceListAccountBuilder {
+    private final GitHub root;
+    private final Requester builder;
+    private final long planId;
+
+    GHMarketplaceListAccountBuilder(GitHub root, long planId) {
+        this.root = root;
+        this.builder = new Requester(root);
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHMarketplaceListAccountBuilder.java]:
```suggestion
        this.builder = root.createRequest();
```

**Comment by PauloMigAlmeida** [comment]
@bitwiseman sure! will do it tomorrow when I get home

File: src/main/java/org/kohsuke/github/GHMarketplaceListAccountBuilder.java (hunk 5 above)
Comment (PauloMigAlmeida) [reply src/main/java/org/kohsuke/github/GHMarketplaceListAccountBuilder.java]: do you mean `root.retrieve()` ?

Comment (PauloMigAlmeida) [reply src/main/java/org/kohsuke/github/GHMarketplaceListAccountBuilder.java]: nevermind :)

**Comment by PauloMigAlmeida** [comment]
@bitwiseman changes implemented. Ready for your review :)

**Pull request merged**